src/PerpetualStableSwap.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+import "./ConditionalOrder.sol";
+import "lib/contracts/src/contracts/GPv2Settlement.sol";
+import "lib/contracts/src/contracts/interfaces/GPv2EIP1271.sol";
+
+// @title A smart contract that is always willing to trade between tokenA and tokenB 1:1,
+// taking decimals into account (and adding specifiable spread)
+contract PerpetualStableSwap is ConditionalOrder, EIP1271Verifier {
+    using GPv2Order for GPv2Order.Data;
+    using SafeMath for uint256;
+    using SafeMath for uint8;
+
+    IERC20 public immutable tokenA;
+    IERC20 public immutable tokenB;
+    uint256 public immutable halfSpreadBps;
+    address public immutable target;
+    bytes32 public domainSeparator;
+
+    // There are 10k basis points in a unit
+    uint256 public constant BPS = 10_000;
+
+    constructor(
+        IERC20 _tokenA,
+        IERC20 _tokenB,
+        uint256 _halfSpreadBps,
+        address _target,
+        GPv2Settlement _settlementContract
+    ) {
+        tokenA = _tokenA;
+        tokenB = _tokenB;
+        halfSpreadBps = _halfSpreadBps;
+        domainSeparator = _settlementContract.domainSeparator();
+
+        if (_target == address(0)) {
```

Comment:
It isn't super clear to me what a non-0 `target` is used for.

Reply:
Hmm, it looks like this is a "receiver", but what isn't clear to me is:
- Why is `ConditionalOrderCreated(_target)` emitted with the receiver (is it supposed to be another `PerpetualStableSwap` order contract?)
- Why the receiver needs an allowance set to the vault relayer.

Reply:
Yes, so the idea is that this contract becomes the fallback handler of a Safe but since the Safe `calls` into the fallback handler (rather than `delegatecalls`) we need to actually check the balances of the target Safe. It's the same pattern that was introduced in for the other order type in #4 

Added a comment to the constructor args

Reply:
Right! It was in the PR description for #4. Thanks for the explanation.

Should this just be `msg.sender` then? `target = 0` doesn't make much sense to me now (especially because there is no way to get the funds out).

Reply:
Bump.

Reply:
The reason I left this in is that potentially you could build custody logic that then subclasses this contract to keep the specific order logic in tact. No strong feelings though as currently there are 0 such use cases.